gap/Parser.gi

```diff
@@ -585,27 +585,30 @@ InstallGlobalFunction( AutoDoc_Parser_ReadFiles,
             if not IsBound( tree!.TitlePage.TitleComment ) then
                 tree!.TitlePage.TitleComment := [ ];
             fi;
-            tree!.content := tree!.TitlePage.TitleComment;
+            current_item := tree!.TitlePage.TitleComment;
 
-            Add( tree, "<URL>" );
-            Add( tree, current_command[ 2 ] );
-            Add( tree, "</URL>" );
+            Add( current_item, "<URL>" );
+            Add( current_item, current_command[ 2 ] );
+            Add( current_item, "</URL>" );
         end
     );
     title_item_list := [ "Title", "Subtitle", "Version", "TitleComment", "Author",
                          "Date", "Address", "Abstract", "Copyright", "Acknowledgements", "Colophon" ];
-    for title_item in title_item_list do
-        command_function_record.( Concatenation( "@", title_item ) ) := function( )
-            current_item := tree;
 
-            if not IsBound( tree!.TitlePage.( title_item ) ) then
-                tree!.TitlePage.( title_item ) := [ ];
+    create_title_item_function := function( name )
+        return function()
+            if not IsBound( tree!.TitlePage.( name ) ) then
+                tree!.TitlePage.( name ) := [ ];
             fi;
-            tree!.content := tree!.TitlePage.( title_item );
-
-            Add( tree, current_command[ 2 ] );
+            current_item := tree!.TitlePage.( name );
+            Add( current_item, current_command[ 2 ] );
         end;
+    end;
+
+    for title_item in title_item_list do
+        command_function_record.( Concatenation( "@", title_item ) ) := create_title_item_function( title_item );
     od;
```

Comment:
Yup. I should have known, I've run into this trap before :-). So, perhaps we could still add a comment to explain this? Something like this: `# Note that we need to create these functions in the helper function create_title_item_function to ensure that the <name> variable is bound properly. Without this intermediate helper, the wrong closure is taken, and the title string for *all* generated functions is set to the same value (namely the value it has at the end of the loop).`